kikuchipy/indexing/_static_dictionary_indexing.py

```diff
@@ -0,0 +1,198 @@
+
+class StaticDictionaryIndexing:
+    """Indexing against pre-computed dictionaries of simulated EBSD
+    patterns.
+    """
+
+    def __init__(self, dictionaries):
+        """Initialize with one or more dictionaries before indexing
+        patterns.
+
+        Parameters
+        ----------
+        dictionaries : EBSD or list of EBSD
+            Dictionaries as EBSD signals with one-dimensional navigation
+            axis and with the `xmap` property set.
+        """
+        if not isinstance(dictionaries, list):
+            dictionaries = [dictionaries]
+        self.dictionaries = dictionaries
+
+    def __call__(
+        self,
+        signal,
+        metric: Union[str, SimilarityMetric] = "zncc",
+        keep_n: int = 1,
+        n_slices: int = 1,
+        return_merged_crystal_map: bool = True,
+        get_orientation_similarity_map: bool = True,
+    ) -> List[CrystalMap]:
+        """Perform dictionary indexing on patterns against preloaded
+        dictionaries, returning a :class:`~orix.crystal_map.CrystalMap`
+        for each dictionary with `scores` and `simulated_indices` as
+        properties.
+
+        Parameters
+        ----------
+        signal : EBSD
+            EBSD signal with experimental patterns.
+        metric : str or SimilarityMetric, optional
+            Similarity metric, by default "zncc".
+        keep_n : int, optional
+            Number of sorted results to keep, by default 1.
+        n_slices : int, optional
+            Number of slices of simulations to process sequentially, by
+            default 1.
+        return_merged_crystal_map : bool, optional
+            Return a merged crystal map, the best matches determined
+            from the similarity scores, in addition to the single phase
+            maps. By default True. See also
+            :func:`~kikuchipy.indexing.merge_crystal_maps`.
+        get_orientation_similarity_map : bool, optional
+            Add orientation similarity maps to the returned crystal maps
+            as an `osm` property, by default True.
+
+        Returns
+        -------
+        xmaps : list of CrystalMap
+            A crystal map for each dictionary loaded and one merged map
+            if `return_merged_crystal_map = True`.
+        """
+        # This needs a rework before sent to cluster and possibly more
+        # automatic slicing with dask
+        num_simulations = self.dictionaries[0].data.shape[0]
+        good_number = 13500
+        if (num_simulations // n_slices) > good_number:
+            answer = input(
+                "You should probably increase n_slices depending on your "
+                f"available memory, try above {num_simulations // good_number}."
+                " Do you want to proceed? [y/n]"
+            )
+            if answer != "y":
+                return
+
+        metric = SIMILARITY_METRICS.get(metric, metric)
+
+        axes_manager = signal.axes_manager
+        spatial_arrays = _get_spatial_arrays(
```

Comment:
There's something wrong with this computation of the spatial arrays, will get this working tomorrow or Wed.

Reply:
This is now OK